### ISSUE TITLE
Id is not required for a Document. The search service can automatically create an identifier

### ISSUE THREAD

**Issue by marcosvidolin** [marcosvidolin/doco#8]
Setting the document identifer is optional. If omitted, the search service will create an identifier.

**Comment by mohammedpithapurwalasc** [comment]
Any updates on this ? This would block us.

**Comment by tj-recess** [comment]
I'd argue that setting ID is rather useful. Could you explain your scenario a little?

Also, if you really don't want a meaningful id then you can work around by not annotating any field with @DocumentId and passing a UUID while creating document. e.g.

```
Document myDoc = doco.toDocument(myEntity, "some-random-id-here");
```

**Comment by mohammedpithapurwalasc** [comment]
Yes, the uuid approach is sufficient. What I wasn't sure of is whether there is a way to specify document id (without requiring an annotation).
